### PR TITLE
Minor tweaks to DELETE to get it ready to add to API-Gateway

### PR DESCRIPTION
**Pull request by Vatsal09**
### Description
[Update]: As discussed, DELETE (soft delete) should just update task(s) deleted value and not return anything

__What__

- [x] tweaked DELETE not to return tasks that were deleted after the operation
- [x] added test for DELETE (soft delete)
- [x] updated gitignore to remove the addition of .DS_Store files
- [x] updated some VSCode settings to make it uniform with other repo's VSCode settings
- [x] added black and pytest to dev-dependencies
- [x] lot of formatting changes for Task resource file

__Why__

To get it ready to connect to api-gateway

__How do we know this works?__

- [x]  unit tests

### CODE REVIEW

**Comment by Vatsal09** [comment]
@joshpaulchan you can finish the review now

**Comment by Vatsal09** [comment]
Test comment for Slack integration

File: resources/Task.py [hunk 1]
```diff
@@ -342,24 +391,22 @@ def delete(self):
...
\ No newline at end of file
+        if not user_id:
+            return error_response("userId missing")
+
+        try:
+            # Soft delete the task(s) by updating the deleted field
+            soft_delete_tasks_by_ids(task_ids)
+            db.session.commit()
+            return success_response([], 204)
+        except:
```
Comment (joshpaulchan) [inline resources/Task.py]: Can you add a logger statement here to log the exception in case it happens?

File: tests/test_task.py [hunk 2]
```diff
@@ -107,8 +117,22 @@ def test_success_response(app):
...
+def test_soft_delete_tasks(app):
+    task_a = Task(id=1, title="Title A", deleted=False)
+    task_b = Task(id=2, title="Title B", deleted=False)
+    save(app, task_a, task_b)
+    soft_delete_tasks_by_ids([1, 2])
+    retrive_task_a = get_task_by_id(1)
+    retrive_task_b = get_task_by_id(2)
+    
+    assert len(get_tasks_by_ids([1,2])) == 0
+    assert retrive_task_a.deleted == True
```
Comment (joshpaulchan) [inline tests/test_task.py]: Should we even return a deleted task in get_task_by_id? I can see arguments for either way. Can you explain your rationale?

File: resources/Task.py [hunk 3]
```diff
@@ -273,57 +312,67 @@ def put(self):
...
-                if ("del_users" in task_data.keys()) and (task_data["del_users"] is not None):
-                    
-                    for user_id in get_users_to_del(task_data.get("taskId"), task_data["del_users"]):
+                    for user_id in get_users_to_del(
+                        task_data.get("taskId"), task_data["del_users"]
+                    ):
                         db.session.delete(user_id)
-                        #task.users.remove(user_id)
-
+                        # task.users.remove(user_id)
```
Comment (joshpaulchan) [inline resources/Task.py]: Suggestion: Remove commented out code

File: resources/Task.py [hunk 4]
```diff
@@ -273,57 +312,67 @@ def put(self):
...
                         db.session.delete(tasklist)
                         # task.tasklists.remove(tasklist)
 
-
                 task_ids_bulk_get.append(task_data.get("taskId"))
                 task_data_bulk_update.append(task)
-            
+
             except Exception as e:
                 print(e)
```
Comment (joshpaulchan) [inline resources/Task.py]: Suggestion: this print should be a logger statement

File: resources/Task.py (hunk 3 above)
Comment (Vatsal09) [reply resources/Task.py]: done

File: tests/test_task.py (hunk 2 above)
Comment (Vatsal09) [reply tests/test_task.py]: Just like the get_tasks_by_ids, get_task_by_id should not return deleted tasks. I exported the logic to a helper function in the test_task.py now. The get_task_by_id will be changed/deleted when I work on the update task next

File: resources/Task.py (hunk 1 above)
Comment (Vatsal09) [reply resources/Task.py]: Added

File: resources/Task.py (hunk 4 above)
Comment (Vatsal09) [reply resources/Task.py]: Yup gotchu. Will add when I add update to gateway.

**Comment by Vatsal09** [comment]
Final check @joshpaulchan?

**Pull request merged**